src/stdlib_math.fypp

```diff
@@ -6,14 +6,155 @@ module stdlib_math
 
     implicit none
     private
-    public :: clip
+    public :: clip, linspace, logspace, lnspace, EULERS_NUMBER_DP, EULERS_NUMBER_SP
+
+    integer, parameter :: DEFAULT_LINSPACE_LENGTH = 100
+    integer, parameter :: DEFAULT_LOGSPACE_LENGTH = 50
+    real(dp), parameter :: DEFAULT_LOGSPACE_BASE = 10
+
+    ! Useful constants for lnspace 
+    real(sp), parameter :: EULERS_NUMBER_DP = 2.71828182845904523536028747135266249775724709369995_sp
+    real(dp), parameter :: EULERS_NUMBER_SP = 2.71828182845904523536028747135266249775724709369995_dp
+    real(qp), parameter :: EULERS_NUMBER_QP = 2.71828182845904523536028747135266249775724709369995_qp
 
     interface clip
     #:for k1, t1 in IR_KINDS_TYPES
         module procedure clip_${k1}$
     #:endfor
     end interface clip
 
+    interface linspace 
+    !! Version: Experimental
+    !!
+    !! Create rank 1 array of linearly spaced elements
+    !! If the number of elements is not specified, create an array with size 100. If n is a negative value,
+    !! return an array with size 0. If n = 1, return end
+    #:for k1, t1 in REAL_KINDS_TYPES
+      #:set RName = rname("linspace_default", 1, t1, k1)
+      module function ${RName}$(start, end) result(res)
+        ${t1}$, intent(in) :: start
+        ${t1}$, intent(in) :: end  
+  
+        ${t1}$ :: res(DEFAULT_LINSPACE_LENGTH)        
+      end function ${RName}$
+    #:endfor
+  
+    #:for k1, t1 in REAL_KINDS_TYPES
+      #:set RName = rname("linspace_n", 1, t1, k1)
+      module function ${RName}$(start, end, n) result(res)
+        ${t1}$, intent(in) :: start
+        ${t1}$, intent(in) :: end  
+        integer, intent(in) :: n
+      
+        ${t1}$ :: res(n)
+      end function ${RName}$
+    #:endfor
+
+    ! Add support for complex linspace
+    #:for k1, t1 in CMPLX_KINDS_TYPES
+      #:set RName = rname("clinspace_default", 1, t1, k1)
+      module function ${RName}$(start, end) result(res)
+        ${t1}$, intent(in) :: start
+        ${t1}$, intent(in) :: end
+
+        ${t1}$ :: res(DEFAULT_LINSPACE_LENGTH)
+      end function ${RName}$
+    #:endfor
+
+    #:for k1, t1 in CMPLX_KINDS_TYPES
+      #:set RName = rname("clinspace_n", 1, t1, k1)
+      module function ${RName}$(start, end, n) result(res)
+        ${t1}$, intent(in) :: start
+        ${t1}$, intent(in) :: end
+        integer, intent(in) :: n
+
+        ${t1}$ :: res(n)
+      end function ${RName}$
+    #:endfor
+
```

Comment:
Could these interfaces be combined with the `real` version, e.g., by defining a fyyp variable `RC_KINDS_TYPES`?

Reply:
The interfaces can definitely get combined. 

The reasoning for this split in interface is because I first implemented a real version, and then later a complex version. Thus the implementations for each function are different but I do believe that the interfaces can get combined. I will combine these two interfaces

Reply:
These interfaces have been meshed :)